### ISSUE TITLE
How to fix the accent issue for languages containing the /r/ sound

### ISSUE THREAD

**Issue by Selectorrr** [Plachtaa/seed-vc#41]

**Comment by Plachtaa** [comment]
Please have a more detailed description of your issue, and the samples of failure cases

**Comment by Plachtaa** [comment]
Thanks for your clear demonstration of the issue, I highly agree with your discovery.
The reason could be follows:
1. Training dataset does not contain language of the source
2. Reference audio is not using the same language as the source

We will release training code in the near future to allow training or finetuning on custom data

**Comment by talipturkmen** [comment]
This is a common problem for models that are not trained with strong R languages. Especially if most of the training data is in English, this problem becomes more apparent. I think training with Italian, Spanish, and Russian would help solve this issue.

Thank you for this great work. I've tested it and it works quite good! 👏 

Just for findings, I foresee three main issues with Seed-VC due to its training and inference scheme:

1. There is accent leakage in cross-lingual voice conversion. Timbre conditioning contains accent and language information. Some adversarial training in this area could help.
2. Conditioning on the target speaker's semantic features leads to style leakage from the prompt, which I think is undesirable for voice conversion. In ideal voice conversion, style and content should come from the source, with only the timbre changing. While conditioning semantic features on the target speaker may improve cloning quality, this style leakage hinders the model's capacity.
3. Maybe training on DAC latents could be better to reach higher quality. 22khz is great but right I think diffusion can do better than that.

**Comment by Plachtaa** [comment]
> This is a common problem for models that are not trained with strong R languages. Especially if most of the training data is in English, this problem becomes more apparent. I think training with Italian, Spanish, and Russian would help solve this issue.
> 
> Thank you for this great work. I've tested it and it works quite good! 👏
> 
> Just for findings, I foresee three main issues with Seed-VC due to its training and inference scheme:
> 
> 1. There is accent leakage in cross-lingual voice conversion. Timbre conditioning contains accent and language information. Some adversarial training in this area could help.
> 2. Conditioning on the target speaker's semantic features leads to style leakage from the prompt, which I think is undesirable for voice conversion. In ideal voice conversion, style and content should come from the source, with only the timbre changing. While conditioning semantic features on the target speaker may improve cloning quality, this style leakage hinders the model's capacity.
> 3. Maybe training on DAC latents could be better to reach higher quality. 22khz is great but right I think diffusion can do better than that.

Thanks for your in-depth analysis. 
However, in my opinion, converted audio mimic target audio is a very desired feature, which makes it resembles target speaker more. This improvements is hard to be measured by Resemblyzer, but very obvious if measuring with WavLM-TDNN or MOS subjective evaluation. Changing only in timbre leads the conversion to be plain an less interesting.
I do not tend to treat your observation as an issue. Next stage of development will be focusing on finding more abstract content representation so that more portion of accent of style can be determined by model itself according to target audio. 
Since the training code and paper is already out, you may try out the solutions you mentioned if keen

**Comment by QinhanJ** [comment]
Same here, the model is so cool, but seems having trouble pronoucing r, rr, in spanish.
Looking forward to see training code!
Such a great project!
